bentoml/_internal/models/model.py

```diff
@@ -173,25 +191,51 @@ def path_of(self, item: str) -> str:
         return self._fs.getsyspath(item)
 
     def flush_info(self):
+        if self._flushed:
+            return
+
         with self._fs.open(MODEL_YAML_FILENAME, "w") as model_yaml:
             self.info.dump(model_yaml)
 
+        self._flushed = True
+
     def flush_custom_objects(self):
+        if self._flushed:
+            return
+
         # pickle custom_objects if it is not None and not empty
         if self.custom_objects:
             with self._fs.open(CUSTOM_OBJECTS_FILENAME, "wb") as cofile:
                 cloudpickle.dump(self.custom_objects, cofile)
 
+        self._flushed = True
```

Comment:
Is it possible that both `custom_objects` and `info` is dirty and we only flush `custom_objects` but marked self flushed, so the `info` part change is lost?